src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -846,7 +846,7 @@ public ProjectState WithMetadataReferences(IReadOnlyList<MetadataReference> meta
         return With(projectInfo: ProjectInfo.With(metadataReferences: metadataReferences).WithVersion(Version.GetNewerVersion()));
     }
 
-    public ProjectState WithAnalyzerReferences(IEnumerable<AnalyzerReference> analyzerReferences)
+    public ProjectState WithAnalyzerReferences(IReadOnlyList<AnalyzerReference> analyzerReferences)
```

Comment:
stronger typing, and this ensures we're not lazily producing these values every time.